### PR TITLE
Heart Disease 2: Electric Boogaloo

### PR DESCRIPTION
**Pull request by ghost**
Up-to-date version of https://github.com/tgstation/tgstation/pull/30870

🆑 Robustin and More Robust Than You
add: A heart disease event has been added. The cure is heart replacement surgery. Effects of cardiac arrest are halted by the chemical Corazone. Once cardiac arrest begins at Stage 5, the disease can be cured by a defibrillator or from a lucky electric shock.
add: Using gym equipment will now grant a hidden exercise buff that prevents heart disease for 20 minutes.
/🆑

### CODE REVIEW

File: code/datums/status_effects/buffs.dm [hunk 1]
```diff
@@ -436,3 +436,8 @@
 	name = "Fleshmend"
 	desc = "Our wounds are rapidly healing. <i>This effect is prevented if we are on fire.</i>"
 	icon_state = "fleshmend"
+
+/datum/status_effect/exercised
```
Comment (vuonojenmustaturska) [inline code/datums/status_effects/buffs.dm]: Status effects process via SSfastprocess, an effect that lasts 20 minutes doesn't need it, please override as necessary

File: code/modules/events/heart_attack.dm [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/datum/round_event_control/heart_attack
+	name = "Random Heart Attack"
+	typepath = /datum/round_event/heart_attack
+	weight = 20
+	max_occurrences = 2
+	earliest_start = 12000
+	min_players = 40 // To avoid shafting lowpop
+
+/datum/round_event/heart_attack/start()
+	for(var/mob/living/carbon/H in shuffle(GLOB.alive_mob_list))
```
Comment (Xhuis) [inline code/modules/events/heart_attack.dm]: just like last time I believe this will make every human player have a heart attack.

**Comment by Durkel** [comment]
This is eventually going to get a nuke op or cuckcultist killed.

**Comment by Robustin** [comment]
Y E S

File: code/modules/events/heart_attack.dm (hunk 2 above)
Comment (Robustin) [reply code/modules/events/heart_attack.dm]: I don't follow, once it gives the disease it breaks the loop.

**Comment by Robustin** [comment]
Also if ops cant get a heart before it hits stage 5 they deserve it, id be far more scared of appendicitis which can have you self-stunning almost immediately. At least with this disease you can attempt to self-cure with shocking.

Also @MoreRobustThanYou I found the var I wanted to add to the original, maybe add one final "check" to the the event's player loop that checks for the mob's "satiety" being under 100, which indicates they've had a lot of junk food. If that condition isn't met then it could just be a 15% probability (continuing, not breaking, if they don't get the disease obviously).

File: code/modules/events/heart_attack.dm (hunk 2 above)
Comment (Xhuis) [reply code/modules/events/heart_attack.dm]: Oh, you're right. This is really weird syntax, but I suppose it works. My bad.
On an unrelated note, using `GLOB.player_list` is probably cleaner here.

**Comment by Alek2ander** [comment]
🤔 Everyone on the station runs all the time, they get plenty of exercise

File: code/modules/events/heart_attack.dm (hunk 2 above)
Comment (MoreRobustThanYou) [reply code/modules/events/heart_attack.dm]: Yeah i'm gonna redo this loop

Comment (Cyberboss) [reply code/modules/events/heart_attack.dm]: you want the intersect of both lists so you only infect alive players

File: code/datums/diseases/heart_failure.dm [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+					to_chat(H, "<span class='warning'>You feel [pick("full", "nauseous", "sweaty", "weak", "tired", "short on breath", "uneasy")].</span>")
+			if(3 to 4)
+				if(!sound)
+					H.playsound_local(H, 'sound/health/slowbeat.ogg',40,0, channel = CHANNEL_HEARTBEAT)
+					sound = TRUE
+				if(prob(3))
+					to_chat(H, "<span class='danger'>You feel a sharp pain in your chest!</span>")
+					if(prob(25))
+						affected_mob.vomit(95)
+					H.emote("cough")
```
Comment (Armhulen) [inline code/datums/diseases/heart_failure.dm]: this will make someone cough and vomit at the same time, which i'm not entirely sure is possible

**Comment by BeeSting12** [comment]
This is a random event right?

**Comment by MoreRobustThanYou** [comment]
yes

File: code/datums/diseases/heart_failure.dm (hunk 3 above)
Comment (TimeIsEndless) [reply code/datums/diseases/heart_failure.dm]: it certainly is in real life

Comment (Armhulen) [reply code/datums/diseases/heart_failure.dm]: i'll take your word on it!

Comment (Spaceinaba) [reply code/datums/diseases/heart_failure.dm]: as somebody with a long history of being sick I can confirm I have coughed and vomited at the same time

**Comment by PKPenguin321** [comment]
Combine it with the appendicitis event and change it to an Organ Failure event?

**Comment by Ordonis** [comment]
Septic shock when

**Pull request merged**